src/learning/nnet.rs

```diff
@@ -328,10 +311,9 @@ impl<'a, T: Criterion> BaseNeuralNet<'a, T> {
             }
         }
 
-        let mut grad = Vec::with_capacity(self.layer_sizes.len() - 1);
-        let mut capacity = 0;
+        let mut gradients = Vec::new();
 
-        for (l, activ_item) in activations.iter().enumerate().take(self.layer_sizes.len() - 1) {
+        for (l, activ_item) in activations[..self.layer_sizes.len() - 1].iter().enumerate() {
```

Comment:
It's fine to leave it as is. Unless [clippy](https://github.com/Manishearth/rust-clippy) complains I'm happy to keep the indexing :)